### ISSUE TITLE
Explore Array Quanitites

### ISSUE THREAD

**Issue by stscijgbot-rstdms** [spacetelescope/romancal#582]
_Issue [RCAL-406](https://jira.stsci.edu/browse/RCAL-406) was created on JIRA by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe):_

Investigate the Impact of Using Quantities for Arrays throughout the Roman calibration pipeline.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=610638#comment-610638):_

Thus far, initial investigations look feasible, however we need to expand ASDF unit support to include the following:

astropy.units.DN

astropy.units.electron (for astropy.units.electron / astropy.units.s)

So that models will serialize.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=616474#comment-616474):_

This is implemented in the following PRs. They are still in draft status while we review them, as this won't be in the next release. These PRs pass all unit tests throughout the entire Roman Calibration Pipeline suite and JWST. Note that the changes for STCAL are the type that will be needed for each reference file that gets turned to quantities in the future.

 

Draft PRs:

STCAL:
spacetelescope/stcal#124

 

RAD:

spacetelescope/rad#192

 

RDM:

spacetelescope/roman_datamodels#111

 

RCAL:
#602

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621295#comment-621295):_

The STCAL PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621297#comment-621297):_

The RAD PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621494#comment-621494):_

The RDM PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621671#comment-621671):_

Work on the RomanCAL PR is done, but it is waiting on new input files before merging.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621295#comment-621295):_

The STCAL PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621297#comment-621297):_

The RAD PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621494#comment-621494):_

The RDM PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621671#comment-621671):_

Work on the RomanCAL PR is done, but it is waiting on new input files before merging.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=626746#comment-626746):_

This was merged and unmerged - the new PR is here: #616

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621295#comment-621295):_

The STCAL PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621297#comment-621297):_

The RAD PR has been merged.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=621671#comment-621671):_

Work on the RomanCAL PR is done, but it is waiting on new input files before merging.

**Comment by stscijgbot-rstdms** [comment]
_Comment by [Paul Huwe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=phuwe) on [JIRA](https://jira.stsci.edu/browse/RCAL-406?focusedCommentId=626746#comment-626746):_

This was merged and unmerged - the new PR is here: #616
